### PR TITLE
K8SPSMDB-265: Disable backups on unmanaged clusters

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade | failed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/752/commits/54226c6374231787d37f4cc80e84f2a84de55743
image: `perconalab/percona-server-mongodb-operator:PR-752-54226c63`

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 1]
```diff
@@ -634,6 +634,10 @@ func (cr *PerconaServerMongoDB) MongosNamespacedName() types.NamespacedName {
 }
 
 func (cr *PerconaServerMongoDB) CanBackup() error {
+	if cr.Spec.Unmanaged {
```
Comment (qJkee) [inline pkg/apis/psmdb/v1/psmdb_types.go:637]: do we need this check there?
We already return error on CheckNSetDefaults

Comment (egegunes) [reply pkg/apis/psmdb/v1/psmdb_types.go:637]: Yes, we do. We're returning the error in `CheckNSetDefaults` if `spec.backup.enabled` is true. User can always create a backup object even if it's false.

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| init-deploy | passed |
| users | passed |
| limits | passed |
| data-sharded | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| non-voting | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/752/commits/54226c6374231787d37f4cc80e84f2a84de55743
image: `perconalab/percona-server-mongodb-operator:PR-752-54226c63`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/752/commits/67e844f5f6d64cc9154f5fac06a013466f3bd150
image: `perconalab/percona-server-mongodb-operator:PR-752-67e844f5`

**Pull request merged**